db/post_migrate/20181251052913_remove_uploaded_meta_id_from_category.rb

```diff
@@ -0,0 +1,10 @@
+class RemoveUploadedMetaIdFromCategory < ActiveRecord::Migration[5.2]
+  def change
+    begin
+      Migration::SafeMigrate.disable!
```

Comment:
`Migration::SafeMigrate` should not be disabled. What you want to do here is to use a post deploy migration instead. https://meta.discourse.org/t/introducing-post-deployment-migration/98974

Reply:
@tgxworld ,I had tried without it but it was failing for me. The file is already in post_migrate folder.

I get this error:
```
== 20181251052914 RemoveUploadedMetaIdFromCategory: migrating =================
-- remove_column(:categories, :uploaded_meta_id, :integer)

WARNING
-------------------------------------------------------------------------------------
An attempt was made to drop or rename a column in a migration
SQL used was: 'ALTER TABLE "categories" DROP COLUMN "uploaded_meta_id"'

Please generate a post deployment migration using `rails g post_migration` to drop
or rename columns.

Note, to minimize disruption use self.ignored_columns = ["column name"] on your
ActiveRecord model, this can be removed 6 months or so later.

This protection is in place to protect us against dropping columns that are currently
in use by live applications.
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Attempt was made to rename or delete column
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:75:in `block (2 levels) in execute'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/dependencies/interlock.rb:48:in `block in permit_concurrent_loads'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:74:in `block in execute'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract_adapter.rb:579:in `block (2 levels) in log'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract_adapter.rb:578:in `block in log'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract_adapter.rb:569:in `log'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:73:in `execute'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract/schema_statements.rb:603:in `remove_column'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:871:in `block in method_missing'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:840:in `block in say_with_time'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:840:in `say_with_time'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:860:in `method_missing'
/home/saurabh/discourse/db/post_migrate/20181251052914_remove_uploaded_meta_id_from_category.rb:3:in `change'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:814:in `exec_migration'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:798:in `block (2 levels) in migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:797:in `block in migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:416:in `with_connection'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:796:in `migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:977:in `migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1292:in `block in execute_migration_in_transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1343:in `block in ddl_transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract/database_statements.rb:259:in `block in transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/connection_adapters/abstract/database_statements.rb:259:in `transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/transactions.rb:212:in `transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1343:in `ddl_transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1291:in `execute_migration_in_transaction'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1263:in `block in migrate_without_lock'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1262:in `each'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1262:in `migrate_without_lock'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1210:in `block in migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1363:in `with_advisory_lock'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1210:in `migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1036:in `up'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/migration.rb:1011:in `migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/tasks/database_tasks.rb:172:in `migrate'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/activerecord-5.2.2/lib/active_record/railties/databases.rake:60:in `block (2 levels) in <main>'
/home/saurabh/.rvm/gems/ruby-2.5.3/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/home/saurabh/.rvm/gems/ruby-2.5.3/bin/ruby_executable_hooks:24:in `eval'
/home/saurabh/.rvm/gems/ruby-2.5.3/bin/ruby_executable_hooks:24:in `<main>'
Tasks: TOP => db:migrate
```

Any idea what I'm doing wrong here?

Reply:
O try `ColumnDropper.execute_to_drop`

Reply:
I had to change from using `change` to `up` - `down` to make it work :sweat_smile: